libs/hmpb/src/vote_matching.ts

```diff
@@ -0,0 +1,59 @@
+import { deepEqual } from '@votingworks/basics';
+import {
+  BallotStyle,
+  Candidate,
+  CandidateContest,
+  getOrderedCandidatesForContestInBallotStyle,
+  GridPositionOption,
+} from '@votingworks/types';
+
+/**
+ * Matches a vote to a grid position by using positional matching.
+ *
+ * For candidate contests with cross-endorsed candidates, the order of grid
+ * positions maps to the order of OrderedCandidateOptions in the ballot style.
```

Comment:
How are we sure that this is true? Do we have any code that enforces that grid positions are in the same order as the ordered candidate options? IIRC, the order of grid positions is currently determined by what order the browser returns the bubble elements in when we are extracting grid layouts from the rendered ballot.

Reply:
Should we consider moving the HMPB bubble position info into the ballot style ordered candidates instead?